### PR TITLE
Zach Campbell

### PR DESCRIPTION
**Pull request by wajnurfes**
- [x] POST
- [x] GET (all)
- [x] GET (specific)
- [x] DELETE
- [x] PUT

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        db.findById(response.id)
+          .then(user => {
+            res.json({ user });
+          });
+      })
+      .catch(error => {
+        res.status(500);
+        res.json({ errorMessage: "There was an error while saving the user to the database" });
+      });
+  }
```
Comment (nphillips78) [inline server.js:31]: Excellent.

File: server.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  });
+});
+
+server.get('/api/users/:id', (req, res) => {
+  const id = req.params.id;
+  db
+    .findById(id)
+    .then(users => {
+      if (users.length) {
+        res.status(200);
```
Comment (nphillips78) [inline server.js:51]: This is an interesting approach. What's your logic in using users.length as a conditional?

File: server.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          res.status(404);
+          res.json({ errorMessage: "The user with the specified ID does not exist." });
+        }
+      })
+      .catch(error => {
+        res.status(500);
+        res.json({ errorMessage: "The user information could not be retrieved." });
+      })
+  }
+});
```
Comment (nphillips78) [inline server.js:94]: Good stuff

**Review by nphillips78 — COMMENTED** [review]
Good work! It's always interesting to see code that uses different logic than my own 😄 Excellent use of conditionals.

File: server.js (hunk 2 above)
Comment (wajnurfes) [reply server.js:51]: Only just now looking at my comments.  If I recall correctly, I discovered that findById was returning an empty array when nothing was found, which makes users.length 0, which evaluates as false.  When a user was found, it was returning an array containing that user, which would have length 1, evaluating as true.  So users.length could be used just like it was a boolean.  It makes my math-brain happy to see equivalencies like that.

**Pull request closed without merging**